r/tests/testthat/test-altrep.R

```diff
@@ -91,11 +91,37 @@ test_that("empty vectors are not altrep", {
   withr::local_options(list(arrow.use_altrep = TRUE))
   v_int <- Array$create(integer())
   v_dbl <- Array$create(numeric())
+  v_str <- Array$create(character())
 
   expect_false(is_arrow_altrep(as.vector(v_int)))
   expect_false(is_arrow_altrep(as.vector(v_dbl)))
+  expect_false(is_arrow_altrep(as.vector(v_str)))
+})
+
+test_that("chunked array become altrep", {
+  s1 <- c("un", "deux", NA)
+  s2 <- c("quatre", "cinq")
+  a <- Array$create(s1)
+  v <- a$as_vector()
+  expect_equal(v, s1)
+  expect_true(is_arrow_altrep(v))
+
+  ca <- ChunkedArray$create(s1, s2)
+  cv <- ca$as_vector()
+  expect_equal(cv, c(s1, s2))
+  expect_true(is_arrow_altrep(cv))
+
+  # chunked array with 2 chunks
+  c_int <- ChunkedArray$create(0L, c(1L, NA, 3L))
+  c_dbl <- ChunkedArray$create(0, c(1, NA, 3))
+  expect_equal(c_int$num_chunks, 2L)
+  expect_equal(c_dbl$num_chunks, 2L)
+
+  expect_true(is_arrow_altrep(as.vector(c_int)))
+  expect_true(is_arrow_altrep(as.vector(c_dbl)))
```

Comment:
Should you also test with a 0-chunk chunked array?

Reply:
✔️